datasette/utils/__init__.py

```diff
@@ -1301,11 +1302,13 @@ def prune_empty_dicts(d: dict):
                 d.pop(key, None)
 
 
-def move_plugins(source: dict, destination: dict):
+def move_plugins(source: dict, destination: dict) -> Tuple[dict, dict]:
     """
     Move 'plugins' keys from source to destination dictionary. Creates hierarchy in destination if needed.
     After moving, recursively remove any keys in the source that are left empty.
     """
+    source = copy.deepcopy(source)
+    destination = copy.deepcopy(destination)
```

Comment:
I'm a bit worried about the performance overhead of this - not on Datasette itself (it only starts up once) but on the Datasette test suite which might call this hundreds of times.

Reply:
I'm going to ignore this problem in the interest of getting this release done.